pkg/router/router_test.go

```diff
@@ -0,0 +1,248 @@
+package router_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	projectfake "github.com/openshift/client-go/project/clientset/versioned/fake"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	routefake "github.com/openshift/client-go/route/clientset/versioned/fake"
+	routelisters "github.com/openshift/client-go/route/listers/route/v1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kubefake "k8s.io/client-go/kubernetes/fake"
+
+	"k8s.io/klog"
+
+	routercmd "github.com/openshift/router/pkg/cmd/infra/router"
+	"github.com/openshift/router/pkg/router"
+	"github.com/openshift/router/pkg/router/controller"
+	templateplugin "github.com/openshift/router/pkg/router/template"
+	"github.com/openshift/router/pkg/router/writerlease"
+)
+
+type harness struct {
+	routeClient routeclient.Interface
+
+	namespace string
+	uidCount  int
+}
+
+func (h *harness) nextUID() types.UID {
+	h.uidCount++
+	return types.UID(fmt.Sprintf("%03d", h.uidCount))
+}
+
+var h *harness
+
+func TestMain(m *testing.M) {
+	logFlags := flag.FlagSet{}
+	klog.InitFlags(&logFlags)
+	if err := logFlags.Set("v", "6"); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Client stuff
+	client := kubefake.NewSimpleClientset()
+	routeClient := routefake.NewSimpleClientset()
+	projectClient := projectfake.NewSimpleClientset()
+
+	namespace := "default"
+
+	h = &harness{
+		routeClient: routeClient,
+		namespace:   namespace,
+	}
+
+	// Other shared junk
+	routerSelection := &routercmd.RouterSelection{}
```

Comment:
If I understand correctly, this will test `hostindex.SameNamespace` but not `hostindex.OldestFirst`.  Do we test with `DisableNamespaceOwnershipCheck: true` too?  Could be a follow-up.

Reply:
You're correct, this tests only the default configuration (i.e. DisableNamespaceOwnershipCheck=false)